src/deckgl-layers/sharedstreets-layer/sharedstreets-layer.js

```diff
@@ -57,22 +73,46 @@ export default class SharedstreetsLayer extends CompositeLayer {
       const {rows: allData, fields} = dataProcessor.processGeobuf(buffer);
       return {
         allData,
-        fields
+        fields,
+        tile: {x, y, z}
       };
     });
   }
 
-  getPickingInfo({info, sourceLayer}) {
-    // info.sourceLayer is the deck gl layer that renders data.
-    const idx = info.sourceLayer.props.idx;
-    const tile = info.tile;
-    const {allData, fields} = tile._data;
-    info.allData = allData;
-    info.fields = fields;
-    info.idx = idx;
-    return info;
+  /**
+   * Called when all tiles in the current viewport are loaded. Calculate and store domains 
+   * for each field, and for each tile.
+   * @param {*} viewportTilesData an array data in the current viewport
+   */
+  onDataLoaded(viewportTilesData) {
+    // data: all data in the viewport => an array of whatever is returned by getTileData
+    if (viewportTilesData && viewportTilesData.length === 0) {
+      return;
+    }
+    const {layers} = this.props;
+    layers && layers.forEach(layer => {
+      for (let i = 0; i < viewportTilesData.length; i++) {
+        const {tile, allData, fields: allDataFields} = viewportTilesData[i];
+        const {x, y, z} = tile;
+        const tileId = `${z}-${x}-${y}`;
+        const data = {
+          allData,
+          fields: allDataFields,
+          filteredIndexForDomain: Object.keys(allData)
+        };  
+        const visualChannels = Object.values(layer.visualChannels);
+        for (let j = 0; j < visualChannels.length; j++) {
+          const channel = visualChannels[j];
+          const field = layer.config[channel.field];
+          if (field) {
+            const newDomain = this.getTileDomain(layer, tileId, field.name, channel, data);
+            this.updateLayerDomain(layer, z, field, channel, newDomain);
```

Comment:
Ideally we can avoid updating a layer domain if this tile has already been merged? This can get pretty expensive depending on the type of the field.